### PR TITLE
Added a one shot execute: Fixes #51

### LINKED ISSUE #51: Provide one-shot methods for execution of single command
motivation: when the user needs to execute only a single command or execute commands that are not depending on each other, the acquire/execute/release dance is not necessary and can be simplified.

```
dbClient.getConnection(ar -> {
  if (ar.failed()) {
  } else {
    SQLConnection connection = ar.result();
    connection.execute(SQL_CREATE_PAGES_TABLE, create -> {
      connection.close();
      if (create.failed()) {
      } else {
      }
     });
  }
});
```

can be simplified to:

```
dbClient.execute(SQL_CREATE_PAGES_TABLE, create -> {
  if (create.failed()) {
  } else {
  }
});
```

changes: provide a version of the current sql commands that operate on the `SQLClient` (the pool) instead of the `SQLConnection`. The commands will operate on the underlying pool instead at the connection level. In the `vertx-jdbc-client` the implementation can benefit from grouping the acquire/execute/release in a single blocking tasks instead of three (it implies a trivial refactor to be able to execute a JDBC command with an existing connection).

### PR DESCRIPTION
**Pull request by pmlopes**

### CODE REVIEW

**Comment by pmlopes** [comment]
See https://github.com/vert-x3/vertx-jdbc-client/pull/106 for examples/tests

**Review by vietj — COMMENTED** [review]
A couple of concerns:

- I think we should find another name than SQLQuery because in the future we might have a query object that represents a query. Also it seems prescriptive for other operations as we want execute or update to be modeled as one-shot query

- why restricted query operations ? we should cover at least the operations used in the wiki application.

**Comment by vietj** [comment]
perhaps `SQLOperations` ?

**Comment by pmlopes** [comment]
I’ve chosen query because in the Postgres documentation query refers to all sql data operations select, update, insert, delete but indeed we’ve been using query as an alias for select and this makes the naming confusing.

Operations can be but for some reason it does not feel good but that’s my opinion. What about just StructuredQuery as it represents the first 2 chars from SQL and is what this interface is all about?

**Comment by vietj** [comment]
maybe simply SQL then ?

File: src/main/java/io/vertx/ext/sql/SQLClient.java [hunk 1]
```diff
@@ -42,11 +44,183 @@
...
-  void close(Handler<AsyncResult<Void>> completionHandler);
+  void close(Handler<AsyncResult<Void>> handler);
+
   /**
    * Close the client
    */
   void close();
+
+  /**
+   * Execute a one shot SQL statement. This method will reduce the boilerplate code by getting a connection from the
```
Comment (vietj) [inline src/main/java/io/vertx/ext/sql/SQLClient.java]: rather:

Execute a single SQL statement, this method acquires a connection from the the pool and (this project), executes the SQL statement and returns it back after the execution.

**Review by vietj — COMMENTED** [review]
It looks good to me, only the javadoc should be changed on the `SQLClient` extension as I explained and be homogeneuous with other methods

**Pull request merged**